SearchInsertPosition

thijsnissen opened the pull request:
Which implementation do you prefer? Find the problem description [here](https://leetcode.com/problems/search-insert-position/). Somehow both implementations are about 100ms slower and use about 10mb more memory than immediately implementing it with a while loop when running the testcases on leetcode's website, even though indexWhere is in fact a while loop behind the scenes...

thijsnissen commented:
Which implementation do you prefer? Find the problem description [here](https://leetcode.com/problems/search-insert-position/). Somehow both implementations are about 100ms slower and use about 10mb more memory than immediately implementing it with a while loop when running the testcases on leetcode's website, even though indexWhere is in fact a while loop behind the scenes...

nmcb reviewed (commented):
this one requires a bit of library knowledge.. but i would prefer:

`nums.indexOf(target)`

:))

which spoils the fun of implementing the exercise ofc <3 and looks a lot like your solution 2 - personally i think the exercise does not require the array to be sorted but states that the implementation can trust the input to be sorted already - anyhow - i would prefer a recursive implementation like:

```
def mySolution(nums: Array[Int], target: Int): Int =
  @tailrec loop(find: Array[Int], idx: Int = 0): Int =
    if      find.isEmpty        then -1
    else if find.head == target then idx
    else    loop(find.tail, idx + 1)


  loop(nums)
```

which also returns `-1` if the target cannot be found

( the latter being more-or-less standard practice )

thijsnissen commented:
Great @nmcb, thanks! I added your solution in as well. indexWhere is used for the same reason I changed your == on line 19 to >=. The problem description states: when the target is not in the array, return on which index it should be added while keeping the array sorted.

Why would prefer a recursive solution compared to using the library functions?
